storage/src/mmr/mutable.rs

```diff
@@ -237,212 +323,341 @@ impl<K: Array, V: Array, H: CHasher> MutableMmr<K, V, H> {
 
         let digests = ops
             .iter()
-            .map(|op| MutableMmr::op_digest(hasher, op))
+            .map(|op| MutableMmr::<_, E, _, _, _>::op_digest(hasher, op))
             .collect::<Vec<_>>();
 
         proof.verify_range_inclusion(hasher, &digests, start_pos, end_pos, root_hash)
     }
 
-    /// Consume the store and return a [StoreState] from which the current active state of the store
-    /// can be fully recovered and fully proven.
-    pub fn to_state(mut self) -> StoreState<K, V, H> {
-        self.prune_known_inactive();
+    /// Commit any pending operations to the store, ensuring they are persisted to disk &
+    /// recoverable upon return from this function.
+    pub async fn commit(&mut self, hasher: &mut H) -> Result<(), Error> {
+        // Raise the inactivity floor by the # of uncommitted operations, plus 1 to account for the
+        // floor op that will be appended.
+        self.raise_inactivity_floor(hasher, self.uncommitted_ops + 1)
```

Comment:
Just commenting that I agree with the idea of treating all ops (including "floor/commit" the same).